py-polars/polars/dataframe/frame.py

```diff
@@ -1085,7 +1087,8 @@ def _div(self, other: Any, *, floordiv: bool) -> DataFrame:
             int_casts = [
                 col(column).cast(tp)
                 for i, (column, tp) in enumerate(self.schema.items())
-                if tp.is_integer() and orig_dtypes[i].is_integer()
+                if tp.is_integer()
+                and (orig_dtypes[i].is_integer() or orig_dtypes[i] == Null)
```

Comment:
This is a fix after something I did broke this test case - https://github.com/pola-rs/polars/blob/2e0438bd8ac675bfbaaf0b027ec3002e461c4776/py-polars/tests/unit/operations/arithmetic/test_arithmetic.py#L287-L288, I have no idea how it was working before though